scripts/cancel-subscriptions/cancel_subscriptions.py

```diff
@@ -0,0 +1,130 @@
+import sys
+import argparse
+from csv import DictReader, DictWriter
+import stripe
+from paypalrestsdk import (
+    BillingAgreement,
+    ResourceNotFound
+)
+
+parser = argparse.ArgumentParser()
+
+parser.add_argument(
+    '--subscription-type',
+    help='The type of subscriptions being cancelled',
+    choices=['stripe', 'paypal'],
+    required=True,
+    dest='type'
+)
+parser.add_argument(
+    '--csv-filename',
+    help='The file where subscriptions are listed',
+    required=True,
+    dest='filename'
+)
+parser.add_argument(
+    '--stripe-key',
+    help='Stripe API key. must be specified if type is "stripe"',
+    dest='stripe_key'
+
+)
+
+args = parser.parse_args()
+
+failed_transactions = []
+
+# Cancels all subscriptions specified in the CSV "filename"
+def cancel_all_subscriptions(filename):
+    print(f'Loading ids from {filename}')
+    ids = load_ids(filename)
+
+    # use the selected cancellation strategy
+    cancellation_strategy = cancel_stripe_subscription if args.type == 'stripe' else cancel_paypal_billing_agreement
+
+    print(f'Cancelling {len(ids)} subscriptions...')
+    for id in ids:
+        cancellation_strategy(id)
+
+    if len(failed_transactions) > 0:
+        print(f'{len(failed_transactions)} failed to cancel, outputting to {args.type}-errors.csv')
+        write_errors(failed_transactions, f'{args.type}-errors.csv')
+        exit(1)
+
+    print('Successfully cancelled all subscriptions!')
+    exit(0)
+
+# Cancel the given Stripe subscription using its ID
+def cancel_stripe_subscription(subscription_id):
+    try:
+        canceled = stripe.Subscription.delete(subscription_id)
+
+        if not canceled or canceled.status != 'canceled':
+            failed_transactions.append(subscription_id)
+
+    except (stripe.error.StripeError, Exception) as e:
+        failed_transactions.append(subscription_id)
+        print(f'Failed to cancel subscription f{subscription_id}. Error: {e}')
+
+# Cancel the given Paypal Billing Agreement using its ID
+def cancel_paypal_billing_agreement(billing_agreement_id):
+    try:
+        billing_agreement = BillingAgreement.find(billing_agreement_id)
+        cancel_note = {'note': 'MZLA migration'}
+
+        if not billing_agreement.id:
+            print(f'The Billing Agreement {billing_agreement_id} was not found.')
+
+        elif billing_agreement.state not in ['Active', 'Suspended']:
+            failed_transactions.append(billing_agreement_id)
+            print(f'Did not cancel Billing Agreement {billing_agreement_id} because it is not Active or Suspended')
+
+        elif not billing_agreement.cancel(cancel_note):
+            failed_transactions.append(billing_agreement_id)
+            print(f'Failed to cancel Billing Agreement {billing_agreement_id}. Error: {billing_agreement.error["message"]}')
+
+    except ResourceNotFound as error:
```

Comment:
Interestingly, not sure this try/except is necessary. I grabbed it from the sample docs for the SDK, but in testing found that it doesn't throw if the ID is fake, it give you a BillingAgreement object with None for most attributes (lol). Doesn't hurt to keep it in case there's an edge case this does catch.